timestamp-impl/src/main/java/com/palantir/timestamp/TimestampAllocationFailures.java

```diff
@@ -32,6 +32,7 @@
     private final Logger log;
 
     private volatile Throwable previousAllocationFailure;
```

Comment:
There might be a bit of a race condition on PAF actually since verify doesn't acquire the lock, so you might get a stale failure rather than the most recent one:

```
responseTo1 lock - read - write PAF1 - unlock - write EMRT
verify read EMRT
responseTo2 lock - read - write PAF2 - unlock - write EMRT
verify read PAF <-- this can be PAF1